### ISSUE TITLE
Python 3.13: test_skip_reason_in_message fails

### ISSUE THREAD

**Issue by musicinmybrain** [nose-devs/nose2#588]
With Python 3.13.0a1:

```
tox -e py313-nocov
```

```
======================================================================
FAIL: test_skip_reason_in_message (nose2.tests.functional.test_junitxml_plugin.JunitXmlPluginFunctionalTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ben/src/forks/nose2/nose2/tests/functional/test_junitxml_plugin.py", line 162, in test_skip_reason_in_message
    self.assertTestRunOutputMatches(
  File "/home/ben/src/forks/nose2/nose2/tests/_common.py", line 82, in assertTestRunOutputMatches
    self.assertRegex(util.safe_decode(cmd_stderr), stderr)
AssertionError: Regex didn't match: 'test \\(test_junitxml_skip_reason.Test\\.test\\) \\.* skip' not found in 'skipped ohai\n\n-----------------------------------------------
[…]
----------------------------------------------------------------------
Ran 327 tests in 19.860s

FAILED (failures=1, skipped=6)
```

It looks like the output has changed to be less verbose.

I know that it’s very early in the Python 3.13 lifecycle, but we’re starting to test with it in Fedora Linux, so I’m sharing what I see. Since it’s not immediately obvious to me what the correct fix is, I’m going to skip this test for now on Python 3.13 builds.

**Comment by sirosen** [comment]
Thanks for reporting this! I'm inclined to wait until betas to tune in to 3.13 , but if this is impacting anyone in a significant way I can start having `nose2` test against it.

**Comment by sirosen** [comment]
I was just able to set aside some time this evening to start digging into this. I can confirm it on 3.12.1 -- I'm able to reproduce.

So far, I've found the cause and am studying/researching resolution.
Under 3.12.0 and lower, `startTest` gets called for skipped tests.

In `nose2`, verbose output only writes part of the description for tests when `startTest` gets called -- vs in `unittest` it seems there's some clever tracking of whether or not that bit of output has been written yet.
The output ends up wonky and this test happens to be the lucky duck to catch it for us.

I've been looking at `unittest` and how it handles things, and can't quite decide on how to proceed yet. There may need to be some more significant adjustments to `nose2` output to align it more with how `unittest` does things.

**Comment by musicinmybrain** [comment]
Thanks for investigating!

**Comment by sirosen** [comment]
I'm pretty sure I've gotten this fixed in `main` (it worked when I tested on 3.12.1), with a relatively narrow solution (#593).

I'm going to wait until the new year to release it, probably January 2nd. If anyone is able to test against `main` and confirm the fix, that would be much appreciated help!

I'll wait to close this until the fix is released, in any case.

**Comment by musicinmybrain** [comment]
Both `tox -e py312-nocov` and `tox -e py312-nocov` work in a checkout of current `main`, using Python 3.12.1 and Python 3.13.0a2, respectively.

Furthermore, there are no test failures in the Fedora Linux [`python-nose2`](https://src.fedoraproject.org/rpms/python-nose2) package on those interpreter versions when I apply https://github.com/nose-devs/nose2/commit/58cbe5ff35e2cc0633e7ff7c46d48a66d597a025.patch to the `0.14.0` release.

It looks to me like #593 is a successful fix! Thank you.

**Comment by sirosen** [comment]
I just realized today that I hadn't released this fix.

I've just published v0.14.1 to pypi with the latest changes. Sorry for the delay, but I hope the fix works well for everyone.

As always, please let me know if you run into any issues using `nose2`!
